tasks/request_files/request_files.py

```diff
@@ -138,19 +139,32 @@ def task(
         )
         retry_sleep_secs = DEFAULT_RESTORE_RETRY_SLEEP_SECS
 
-    # Get retrieval type
+    # Get retrieval type value from config first
     try:
-        retrieval_type = os.environ[OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY]
-        if retrieval_type not in ("Standard", "Bulk", "Expedited"):
-            msg = (
-                f"Invalid RESTORE_RETRIEVAL_TYPE: '{retrieval_type}'"
-                " defaulting to 'Standard'"
-            )
-            LOGGER.info(msg)
-            retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
+        retrieval_type = event["config"][CONFIG_RESTORE_RETRIEVAL_TYPE_KEY]
+        LOGGER.info(f"Found the following retrieval type from config: {retrieval_type}")
     except KeyError:
-        LOGGER.warn("Invalid RESTORE_RETRIEVAL_TYPE: 'None' defaulting to 'Standard'")
-        retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
+        LOGGER.warning(
+            f"{CONFIG_RESTORE_RETRIEVAL_TYPE_KEY} is not set. Using RESTORE_RETRIEVAL_TYPE environment value."
+        )
+        retrieval_type = None
+
+    if retrieval_type is None or retrieval_type not in ("Standard", "Bulk", "Expedited"):
```

Comment:
With the `except` above, this could result in two messages for the same error. Recommend removing the first message. That also lets you use `event["config"].get(CONFIG_RESTORE_RETRIEVAL_TYPE_KEY, None)` and drop the `try/except` since this line handles errors.

Reply:
I concur with @AndrewDorn, lines 141 to 166 could be simplified down and values can be defaulted to none for easy checking. Also consider putting the valid restore types in a list variable for easy checking:

```
VALID_RESTORE_TYPES = ["Bulk", "Expedited", "Standard"] 

# Get config and env set to None if the key does not exist
env_retrieval_type = os.getenv(OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY, None)
config_retrieval_type = event["config"].get(CONFIG_RESTORE_RETRIEVAL_TYPE_KEY, None)

# Set initial default to deployment env value
retreival_type = env_retrieval_type

# Overide with config if we are able
if config_retrieval_type is not None:
    if config_retrieval_type in VALID_RESTORE_TYPES:
        retreival_type = config_retrieval_type
        ... Logging or other logic
       
 # Safety catch in case someone put a bogus value in env
if (retreival_type is None) or (retreival_type not in VALID_RESTORE_TYPES):
    retreival_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
    ... Some logging or other logic
```  

I would keep the logging to let the user know that we are setting values because whatever they set is unsafe.

Reply:
I would prefer we raise errors when an illegal value is detected. This also processes out-of-order (env default -> config override -> env default -> config override -> env default -> code default), which makes maintenance/reading difficult.
```
# Look for config override
retrieval_type = event["config"].get(CONFIG_RESTORE_RETRIEVAL_TYPE_KEY, None)
if retrieval_type is not None:
	if retrieval_type in VALID_RESTORE_TYPES:
		LOGGER.info(f"Found glacier restore type from config: {retrieval_type}. Overridding other values.")
	else:
		# Log and raise error

# Look for default from TF
if retrieval_type is None:
	retrieval_type = os.getenv(OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY, None)
		if retrieval_type is not None:
			if retrieval_type in VALID_RESTORE_TYPES:
				LOGGER.info(f"Found restore type from {OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY}: {retrieval_type}")
			else:
				# Log and raise error

# Use fallback, which should not be needed due to default TF value
if retrieval_type is None:
	retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
	LOGGER.info(f"defaulting to {DEFAULT_RESTORE_RETRIEVAL_TYPE}")
```

Reply:
seems like we now have a few different approach. If we want to raise errors, I say we use the above that Andrew posted.

Reply:
added more logging and added more unit tests